Packs/TAXIIServer/Integrations/TAXIIServer/TAXIIServer.py

```diff
@@ -281,15 +288,24 @@ def yield_response() -> Generator:
             mimetype='application/xml'
         )
 
-    def get_url(self):
+    def get_url(self, request_headers: Headers) -> str:
         """
+        Args:
+            request_headers: The request headers
+
         Returns:
             The service URL according to the protocol.
         """
-        if self.http_server:
-            return f'{self.host}:{self.port}'
+        if request_headers and '/instance/execute' in request_headers.get('X-Request-URI', ''):
+            # if the server rerouting is used, then the X-Request-URI header is added to the request by the server
+            # and we should use the /instance/execute endpoint in the address
+            self.url_scheme = 'https'
+            calling_context = get_calling_context()
+            instance_name = calling_context.get('IntegrationInstance', '')
```

Comment:
How do we deal here with an instance name that has spaces or special chars that require url encoding?

Reply:
@Itay4 I think we need to URL encode here?

Reply:
yea added it in the line below